float64 x
float64 y
int32 flag
float64 speed
int32 curve
